plugins/flytekit-data-fsspec/flytekitplugins/fsspec/pandas.py

```diff
@@ -58,14 +58,13 @@ def decode(
         self,
         ctx: FlyteContext,
         flyte_value: literals.StructuredDataset,
+        current_task_metadata: typing.Optional[StructuredDatasetMetadata] = None,
     ) -> pd.DataFrame:
         uri = flyte_value.uri
         columns = None
         kwargs = get_storage_options(uri)
-        if flyte_value.metadata.structured_dataset_type.columns:
-            columns = []
-            for c in flyte_value.metadata.structured_dataset_type.columns:
-                columns.append(c.name)
+        if current_task_metadata and current_task_metadata.structured_dataset_type.columns:
```

Comment:
If there's no columns here then columns in the read_parquet function will be `None`.  Will that work okay?  It'll pull all columns right?

Reply:
yeah, will read all the column if `columns` value is None